connect-inject/container_init.go

```diff
@@ -85,6 +96,35 @@ func (h *Handler) containerInit(pod *corev1.Pod, k8sNamespace string) (corev1.Co
 		}
 	}
 
+	enableMetrics, err := h.enableMetrics(pod)
```

Comment:
Are there common cases where `enableMetrics` is true but `run` is not?
If so, maybe we could document that somewhere in the code?
If not, you could potentially just wrap everything inside `if run`. 🤷🏽

Reply:
Yeah the case where you'd want `data.EnableMetrics` and `data.PrometheusScrapeListener` to be set but not the values in `if run` is if your app doesn't have any metrics and you still want Envoy sidecar metrics. So you'd enable metrics, but not metrics merging, which means prometheus annotations are still added to your app, and envoy_prometheus_bind_addr gets configured, but the merged metrics server wouldn't be running. I'll document that in a comment, that's a good thing to call out, thanks for catching it!